nextflow/src/org/labkey/nextflow/pipeline/NextFlowPipelineJob.java

```diff
@@ -0,0 +1,60 @@
+package org.labkey.nextflow.pipeline;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.labkey.api.pipeline.PipeRoot;
+import org.labkey.api.pipeline.PipelineJob;
+import org.labkey.api.pipeline.PipelineJobService;
+import org.labkey.api.pipeline.TaskId;
+import org.labkey.api.pipeline.TaskPipeline;
+import org.labkey.api.util.FileUtil;
+import org.labkey.api.util.URLHelper;
+import org.labkey.api.util.UnexpectedException;
+import org.labkey.api.view.ViewBackgroundInfo;
+
+import java.io.IOException;
+
+public class NextFlowPipelineJob extends PipelineJob
+{
+    private String _apiKey;
+    // For serialization
+    protected NextFlowPipelineJob()
+    {}
+
+    public NextFlowPipelineJob(ViewBackgroundInfo info, @NotNull PipeRoot root, String apiKey)
+    {
+        super(null, info, root);
+        this._apiKey = apiKey;
+        try
+        {
+            setLogFile(FileUtil.createTempFile("NextFlowPipelineJob", ".log"));
```

Comment:
This shouldn't be a temp file. Create a file based on the current date/time in the pipeline root. See `MaintenancePipelineJob` for an example.

Reply:
done